src/components/tabs/index.tsx

```diff
@@ -0,0 +1,85 @@
+import classNames from 'classnames';
+import React, {
+    Children,
+    cloneElement,
+    useCallback,
+    useState,
+    useEffect,
+} from 'react';
+import { Tab } from './tab';
+
+import './index.scss';
+
+interface TabsProps {
+    children: JSX.Element[];
+    className?: string;
+    selectedTabIndex?: number;
+    contentClassName?: string;
+    contentPanelClassName?: string;
+}
+
+const Tabs: React.FC<TabsProps> = ({
+    children,
+    className,
+    selectedTabIndex,
+    contentClassName,
+    contentPanelClassName,
+}) => {
+    const [activeTabIndex, setActiveTab] = useState(0);
+
+    useEffect(() => {
+        if (
+            selectedTabIndex &&
+            selectedTabIndex > -1 &&
+            children[selectedTabIndex]
+        ) {
+            setActiveTab(selectedTabIndex);
+        }
+    }, []);
+
+    const handleClick = useCallback(
+        (index: number) => {
+            if (index !== activeTabIndex) {
+                setActiveTab(index);
+            }
+        },
+        [activeTabIndex]
+    );
+
+    const renderChildrenWithTabsApiAsProps = useCallback(() => {
+        return Children.map(children, (child, index) => {
+            return cloneElement(child, {
+                tabIndex: index,
+                active: index === activeTabIndex,
+                onClick: handleClick,
+            });
+        });
+    }, [children[activeTabIndex]]);
+
+    const renderActiveTabContent = useCallback(() => {
+        if (children[activeTabIndex]) {
+            return children[activeTabIndex].props.panel;
+        }
+    }, [children[activeTabIndex]]);
+
+    return (
+        <>
```

Comment:
The root tag should contain main class of the component to have an impact on all children